shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/product_details/ProductDetailsViewModel.kt

```diff
@@ -47,14 +55,42 @@ class ProductDetailsViewModel(
         }
     }
 
-    fun onWantClicked() {
+    fun onWantClicked(
+        productDetailsOpenedFrom: ProductDetailsOpenedFrom,
+    ) {
         menuProductDetailsState.value.menuProduct?.let { menuProduct ->
+            sendOnWantedClickedAnalytic(
+                menuProductUuid = menuProduct.uuid,
+                productDetailsOpenedFrom = productDetailsOpenedFrom
+            )
             sharedScope.launch {
                 addCartProductUseCase(menuProduct.uuid)
             }
         }
     }
 
+    private fun sendOnWantedClickedAnalytic(
+        menuProductUuid: String,
+        productDetailsOpenedFrom: ProductDetailsOpenedFrom,
+    ) {
+        when (productDetailsOpenedFrom) {
+            ProductDetailsOpenedFrom.RECOMMENDATION_PRODUCT -> analyticService.sendEvent(
+                RecommendationAddEvent,
+                params = listOf(EventParameter("menuProductUuid", menuProductUuid))
+            )
+
+            ProductDetailsOpenedFrom.CART_PRODUCT -> analyticService.sendEvent(
+                CartAddEvent,
+                params = listOf(EventParameter("menuProductUuid", menuProductUuid))
```

Comment:
- Let's add `EventParameter` child for all concrete param like `MenuProductUuidEventParameter` to be sure that all keys will be valid
- Let's add these params right to concrete events like `class CartAddEvent(menuProductUuidEventParameter: MenuProductUuidEventParameter)` to be sure that we will pass all necessary params